src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -30,15 +30,21 @@
  * Authors: 2016 Pan Deng, Soumyajit De, Heiko Strathmann, Viktor Gal
  */
 
+#ifndef LINALG_NAMESPACE_H_
+#define LINALG_NAMESPACE_H_
+
 #include <shogun/mathematics/linalg/LinalgBackendBase.h>
 #include <shogun/mathematics/linalg/SGLinalg.h>
+#include <shogun/lib/Lock.h>
 
 namespace shogun
 {
 
 namespace linalg
 {
 
+static CLock m_gpu_transfer;
```

Comment:
@karlnapf why do you say that `CLock`? i mean yeah in this branch which is not rebased for ages it is still that but latest develop branch has better backend for locking